src/core/vector/qgsvectordataprovider.cpp

```diff
@@ -936,6 +936,18 @@ QgsGeometry QgsVectorDataProvider::convertToProviderType( const QgsGeometry &geo
     }
   }
 
+  //convert circularstring to compoundcurve
+  if ( QgsWkbTypes::flatType( providerGeomType ) == Qgis::WkbType::CompoundCurve )
+  {
+    QgsCircularString *circularString = qgsgeometry_cast<QgsCircularString *>( geometry );
+    if ( circularString )
```

Comment:
Should we also handle linestring -> compound curve auto translation here? Or is that done elsewhere?